components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -726,9 +735,27 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
      * @param oAuthMessage      oAuth Message.
      * @return                  The consent url.
      */
+    @Deprecated
     public static String getUserConsentURL(OAuth2Parameters params, String loggedInUser, String sessionDataKey,
                                            boolean isOIDC, OAuthMessage oAuthMessage) throws OAuthSystemException {
 
+        return getUserConsentURL(params, loggedInUser, sessionDataKey, isOIDC, null, StringUtils.EMPTY);
+    }
+
+    /**
+     * Returns the consent page URL.
+     *
+     * @param params                OAuth2 Parameters.
+     * @param loggedInUser          The logged in user
+     * @param isOIDC                Whether the flow is an OIDC or not.
+     * @param oAuthMessage          oAuth Message.
+     * @param additionalQueryParams Additional query params to be appended to the consent page url.
+     * @return                      The consent url.
+     */
+    public static String getUserConsentURL(OAuth2Parameters params, String loggedInUser, String sessionDataKey,
+                                           boolean isOIDC, OAuthMessage oAuthMessage, String additionalQueryParams)
+            throws OAuthSystemException {
```

Comment:
Can we review these parameters and see whether all of these needs to be passed from the callee?

Reply:
Resolved by https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2020/commits/a4780e126be8e00ac7e209a2553485da861c8cfb